### PR TITLE
implemented support for JWT, added login endpoint

### PR DESCRIPTION
**Pull request by citruscaviar**

### CODE REVIEW

File: src/main/java/com/onlinebookstore/store/config/SecurityConfig.java [hunk 1]
```diff
@@ -34,16 +38,25 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
             .csrf(AbstractHttpConfigurer::disable)
             .authorizeHttpRequests(
                 auth -> auth
-                    .requestMatchers("/auth/**",
+                    .requestMatchers("/auth/registration", "/auth/login",
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/config/SecurityConfig.java:41]:
```suggestion
                    .requestMatchers("/auth/**",
```

File: src/main/java/com/onlinebookstore/store/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                        null,
+                        userDetails.getAuthorities()
+                );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/security/JwtAuthenticationFilter.java:47]:
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/com/onlinebookstore/store/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/security/JwtAuthenticationFilter.java:48]: create constant for `TOKEN_HEADER = "Bearer "`

File: src/main/java/com/onlinebookstore/store/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/security/JwtAuthenticationFilter.java:49]:
```suggestion
            return bearerToken.substring(TOKEN_HEADER.length());
```

File: src/main/resources/db/changelog/changes/05-create-users_roles-table.yml [hunk 5]
```diff
@@ -22,4 +22,6 @@ databaseChangeLog:
                     referencedTableName: roles
                     referencedColumnNames: id
                     nullable: false
-                    primaryKey: true
+        - addPrimaryKey:
+            columnNames: role_id, user_id
+            tableName: users_roles
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/05-create-users_roles-table.yml:27]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by liliia-ponomarenko — APPROVED** [review]
Awesome 😎

**Pull request merged**